### PR TITLE
Support cross compiling pocl

### PR DESCRIPTION
**Pull request by isuruf**
I'm trying to cross compile for macos-arm64 from macos-x86_64.

### CODE REVIEW

**Comment by pjaaskel** [comment]
This needs to be tested on an FPGA SoC with an LLVM-less build (the original use case in our group).

**Comment by pjaaskel** [comment]
@topileppanen can you test that we can still cross-build an LLVM-less for PYNQ after this is applied?

File: cmake/FindHwloc.cmake [hunk 1]
```diff
@@ -134,34 +134,10 @@ if(WIN32)
...
-    set(Hwloc_VERSION "1.7.0")
-  endif()
-
-  else() # Find with pkgconfig for non-crosscompile builds
-
   find_package(PkgConfig)
 
   if(HWLOC_ROOT)
-    set(ENV{PKG_CONFIG_PATH} "${HWLOC_ROOT}/lib/pkgconfig")
+    set(ENV{PKG_CONFIG_PATH} "{HWLOC_ROOT}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
```
Comment (isuruf) [inline cmake/FindHwloc.cmake:140]:
```suggestion
    set(ENV{PKG_CONFIG_PATH} "${HWLOC_ROOT}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
```

**Comment by keryell** [comment]
I am curious about what you are doing with PYNQ.
Are you using only the ARM cores or also the FPGA?

**Comment by pjaaskel** [comment]
We don't really use the ARM cores except for comparisons. CPC researches portable ways to utilize FPGAs from higher-level languages. One smaller project is the pocl-accel work https://www.computer.org/publications/tech-news/accelerator-framework-for-portable-computing-language which is still ongoing, another is the longer term research track of ours where we look into using TTA as a light weight but scalable OpenCL-programmable overlay for making FPGAs as "easy" to use as GPUs. E.g. https://www.researchgate.net/publication/346203698_TTA-SIMD_Soft_Core_Processors If you know of workloads that can be made competitive on FPGAs in general, please let me know. It's a bit hard to find those.

**Comment by pjaaskel** [comment]
@isuruf  can you rebase this to a single commit which is easier to revert if we'll find problems in our cases?

**Pull request merged**